pkg/engine/combat/damage.go

```diff
@@ -0,0 +1,65 @@
+package combat
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+func (mgr *Manager) totalDamage(h *info.Hit, base float64, dmg float64, crit bool) float64 {
+	def := h.Defender.DEF()
+	def_mult := 1 - (def / (def + 200 + 10*float64(h.Attacker.Level())))
+
+	res := mgr.res(h)
+	vul := mgr.vul(h)
+
+	// ByPureDamage equation does not scale on DMG%, and break effect applies.
+	breakDmg := 1.0
+	if h.AttackType == model.AttackType_ELEMENT_DAMAGE {
+		breakDmg += float64(h.Attacker.BreakEffect())
+		dmg = 1.0
+	}
+
+	// toughness multiplier
+	toughness_multiplier := 0.9
+	if h.Defender.Stance() == 0 {
+		toughness_multiplier = 1
+	}
+
+	fatigue := 1 - h.Attacker.GetProperty(prop.Fatigue)
+	AllDamageReduce := h.Defender.GetProperty(prop.AllDamageReduce)
+	if AllDamageReduce < 0.01 {
+		AllDamageReduce = 0.01
+	}
```

Comment:
`AllDamageReduce` needs to be `(1 - AllDamageReduce)` when used in formula. In current state, 0% `AllDamageReduce` will result in 0 damage done